runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -514,6 +514,13 @@ TR_J9ServerVM::isPrimitiveClass(TR_OpaqueClassBlock * clazz)
 bool
 TR_J9ServerVM::isClassInitialized(TR_OpaqueClassBlock * clazz)
    {
+   OMR::CriticalSection getRemoteROMClass(_compInfoPT->getClientData()->getROMMapMonitor());
+   auto it = _compInfoPT->getClientData()->getROMClassMap().find((J9Class*) clazz);
```

Comment:
If the class is not initialized we don't want to cache and return that answer forever. Instead a new message should be sent to the client in the hope that by now the class has been initialized. So the code would look something like:
```
check cache
if class is cached
   if class is initialized
      return true
    else
        send message
        cache answer
        return answer
else
    send message
    return answer
```
Make sure you don't hold the lock while sending the message